certification/internal/engine/podman.go

```diff
@@ -177,3 +149,122 @@ func (p *podmanEngine) WaitContainer(containerId string, waitOptions cli.WaitOpt
 	log.Info("container reached a running state")
 	return true, nil
 }
+
+func (p *podmanEngine) RunSystemContainer(containerName string) (*cli.PodmanOutput, error) {
+	log.Debug(fmt.Sprintf("Generating Systemd file for container %s", containerName))
+	cmdArgs := []string{"generate", "systemd", "--new", "--files", "--name", containerName}
+
+	output, err := RunCommand("podman", cmdArgs)
+
+	if err != nil {
+		return output, err
+	}
+	serviceFilepath := strings.TrimSpace(output.Stdout)
+
+	r, err := regexp.Compile(".*/(.+.service)")
+	if err != nil {
+		log.Error("unable to fetch the name of the systemd service: ", err)
+		return &cli.PodmanOutput{
+			Stderr: err.Error(),
+		}, err
+	}
+	serviceName := r.FindStringSubmatch(serviceFilepath)[1]
+
+	// copy the unit file from working dir to systemd dir
+	_, err = copyFile(serviceFilepath, fmt.Sprintf("%s/%s", syetemdDir, serviceName))
+
+	// remove the generated unit file from the working dir
+	defer os.Remove(serviceFilepath)
+
+	if err != nil {
+		log.Error(fmt.Sprintf("unable to copy the unit file into the systemd dir %s: ", syetemdDir), err)
+		return &cli.PodmanOutput{
+			Stderr: err.Error(),
+		}, err
+	}
+
+	log.Debug(fmt.Sprintf("Reloading daemon set and start the service %s", serviceName))
+
+	if output, err = RunCommand("systemctl", []string{"daemon-reload"}); err != nil {
+		log.Error("unable to reaload the daemon set: ", err)
+		log.Error("Stderr: ", output.Stderr)
+		return &cli.PodmanOutput{
+			Stderr: output.Stderr,
+		}, err
+	}
+	if output, err = RunCommand("systemctl", []string{"start", serviceName}); err != nil {
+		log.Error(fmt.Sprintf("unable to start the service %s: ", serviceName), err)
+		log.Error("Stderr: ", output.Stderr)
+		return &cli.PodmanOutput{
+			Stderr: output.Stderr,
+		}, err
+	}
```

Comment:
@bcrochet @acornett21 @tonytcampbell @jomkz Since this check needs to copy unit file into systemd dir, and start/stop services, the preflight binary needs to run as root. Is it ok?

Reply:
I think I'd rather us call sudo, rather than expect the binary to run as root. Then, it is up to the caller of preflight to ensure that sudo is set up correctly, but it also gives them the opportunity to limit what preflight has access to.

Reply:
I hadn't thought of @bcrochet 's approach, but that makes sense, and puts us more in control of when to execute `sudo` and when not to.

Reply:
Great point! I updated the PR to only run with `sudo` if it is absolutely necessary. Please let me know if you have any feedbacks.